### PR TITLE
Update AWS SDK to latest

### PR DESCRIPTION
**Pull request by notnoop**
Update to the latest AWS SDK to pick up EC2 Instance Metadata Service v2
support, announced in Nov 19, 2019.  Updating it here is necessary for
downstream projects that haven't adopted go mod (e.g. Nomad), and rely
on vendored provider packages here.

For more info on IMDSv2:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html
.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,11 +5,12 @@ require (
 	github.com/Azure/azure-sdk-for-go v16.0.0+incompatible
 	github.com/Azure/go-autorest v10.7.0+incompatible
 	github.com/abdullin/seq v0.0.0-20160510034733-d5467c17e7af // indirect
-	github.com/aws/aws-sdk-go v1.15.24
+	github.com/aws/aws-sdk-go v1.15.41
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/denverdino/aliyungo v0.0.0-20170926055100-d3308649c661
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible // indirect
 	github.com/digitalocean/godo v1.1.1
+	github.com/dimchansky/utfbom v1.1.0 // indirect
```
Comment (notnoop) [inline go.mod]: This is unrelated change but it looks like it's a dependency of `github.com/Azure/go-autorest` v.10.7.0 [1] but somehow we missed declaring it here.

[1] https://github.com/Azure/go-autorest/blob/v10.7.0/autorest/azure/auth/auth.go#L34

Comment (banks) [reply go.mod]: Is this the most conservative update that includes the things we need?

FWIW Consul now pulls in `1.25.42` anyway thanks to recent AWS CA provider work and we tested go-discover integration afterwards and it works.

Comment (notnoop) [reply go.mod]: Oops - I had updated it to `v1.25.41` the most recent release with `ec2metadata` changes[1].  I accidentally updated it with `v1.15.41` earlier.

[1] https://github.com/aws/aws-sdk-go/releases/

Comment (banks) [reply go.mod]: Correction: Consul pulls in `v1.25.32` which is what I tested with go-discover stuff.

Let me sanity check Consul AWS cloud auto join with 1.25.41 and then we can call this good.

Comment (banks) [reply go.mod]: Yep works with 1.25.41:

```bash
$ go get github.com/aws/aws-sdk-go@v1.25.41
...
$ make dev
$ envchain csl consul agent -retry-join "provider=aws tag_key=Role tag_value=consul-server region=us-west-1" -data-dir "/tmp/consul"
...
2019/12/02 15:26:47 [INFO] agent: Discovered LAN servers: 10.0.1.241 10.0.1.174 10.0.1.12 10.0.1.33 10.0.1.237 10.0.1.44 10.0.1.11 10.0.1.156 10.0.1.28
...
```

(We have dev servers running in that region so this discovered those and could join them.

**Comment by banks** [comment]
I also ran our integration tests for the new CA using this version for completeness and got a pass:

```
$ envchain csl go test -v github.com/hashicorp/consul/agent/connect/ca -count=1 -run='^TestAWS'
...
PASS
ok  	github.com/hashicorp/consul/agent/connect/ca	37.372s
```

**Comment by notnoop** [comment]
Thanks!

**Pull request merged**